### ISSUE TITLE
Compromise-dates plugin mutates context param obj on function calls

### ISSUE THREAD

**Issue by Fdawgs** [spencermountain/compromise#1109]
Node version: 20.13.1
Compromise version: 14.13.0
Compromise-dates version:  3.5.0

Using the compromise-dates plugin, any call to the `.dates()` [functions](https://github.com/spencermountain/compromise/tree/master/plugins/dates#api) (`.json()`, `.get()`, `.format()` etc.) mutates the `today` value in the context param obj to a SpaceTime object, which is an unexpected side-effect:

```js
'use strict'

/** @type {import('compromise').default} */
const nlp = require('compromise')
const nlpDates = require('compromise-dates')
nlp.plugin(nlpDates)

const doc = nlp("I'll deal with that bug tomorrow")

const context = {
  timezone: 'Canada/Eastern',
  today: '2020-02-20',
  punt: { weeks: 2 },
  dayStart: '8:00am',
  dayEnd: '5:30pm',
}

const datesViews = doc.dates(context)
// context obj is still the same at this point
console.log(context)

const dateList = datesViews.json()

console.log(context)
/* context obj is now:
{
  timezone: 'Canada/Eastern',
  today: SpaceTime {
    epoch: 1582174800000,
    tz: 'canada/eastern',
    silent: true,
    british: undefined,
    _weekStart: 1,
    _today: {}
  },
  punt: { weeks: 2 },
  dayStart: '8:00am',
  dayEnd: '5:30pm'
}
 */
```

**Comment by spencermountain** [comment]
Hey! Good catch - .json should definitely never mutate the object - getting swamped at work, will look at a fix hopefully this weekend
Cheers

**Comment by Fdawgs** [comment]
Thanks @spencermountain, no worries! Just to reiterate, it seems to be all/most of the `.dates()` functions. i.e.
`doc.dates(context).format('{month} {date-ordinal}')` also mutates it.

**Comment by spencermountain** [comment]
hey Frazer, think this is fixed in the 3.6.0 release of compromise-dates. Please let me know if you see any other funny-biz
thanks

**Comment by Fdawgs** [comment]
Great, thanks Spencer!
